main/gradle/src/mill/main/gradle/JavaModel.java

```diff
@@ -0,0 +1,125 @@
+package mill.main.gradle;
```

Comment:
Where did these model files come from? Were they copy-pasted from somethird-party code, or did you come up with them from scratch? If you came up with them, did you reference any documentation or code when writing them? We should leave a comment so people can see where these model files came from

Reply:
All the models were written from scratch.
- Gradle requires every model to be a `Serializable` interface (or a runtime error is thrown).
- The relevant plugins/settings were identified by searching the Gradle [source repository](https://github.com/gradle/gradle/tree/master).
- Tooling API usage was adapted from an old [demo project](https://github.com/radimk/gradle-toolingApi-demo).
- Latest Gradle API was obtained after posting on the [forum](https://discuss.gradle.org/t/unable-to-locate-latest-gradle-api-libraries/50157/2).

I have added some docs and links but there isn't much information on the internals of the Gradle API.

@ShreckYe Did you find any documentation that could be useful here?

Reply:
I guess linking to the Javadocs of `gradle.api.Project`, `org.gradle.api.artifacts.Configuration`, `org.gradle.api.artifacts.Dependency`, etc. should be enough. I can't find other types of tutorials or more elaborate documentation either.